sessions/js-array-methods-2/animal-strings/index.js

```diff
@@ -0,0 +1,63 @@
+console.clear();
+
+const animalStrings = [
+	'cat',
+	'dog',
+	'elephant',
+	'rabbit',
+	'lion',
+	'tiger',
+	'horse',
+	'giraffe',
+	'zebra',
+	'penguin',
+	'polar bear',
+	'panda',
+	'koala',
+	'kangoroo',
+	'monkey',
+	'sloth',
+	'hippo',
+	'rhino',
+];
+
+const hippoExists = null;
+
+// Hint: You can pass the starting index as second parameter.
+const catStartingFromIndexFiveExists = null;
+
+// Hint: Besides the array method, check out the string method `startsWith()`.
+const firstAnimalStartingWithLetterP = null;
+
+const indexOfGiraffe = null;
+
+// Note:
+// - Sorting strings is slightly more complicated than sorting numbers.
+// - You will need if-statements and return values of -1, 1, and 0.
+// - Check out MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort
+
+// Hint: There is no need to upper-/lowercase the strings before sorting them.
+// Hint: sort() mutates the original array, which is bad.
+// -> Use animals.slice().sort(...) to make a copy.
+const animalsSortedAlphabetically = null;
+
+// Hint: Guess what? There is a string method called `endsWith()`.
+const anyAnimalEndsWithLetterZ = null;
+
+const everyAnimalHasMoreThanTwoLetters = null;
+
+const valuesForLengthOfAllAnimalNames = [3, 3, 8, 6, 4, 5, 5, 7, 5, 7, 10, 5, 5, 8, 6, 5, 5, 5];
+
+// Hint: Use the variable 'valuesForLengthOfAllAnimalNames' to sum up all letters.
+const sumOfAllAnimalLetters = null;
```

Comment:
// The Solution with the suggested change could look like this:
```js
const sumOfAllAnimalLetters = animalStrings.reduce(
  (previousValue, currentValue) => {
    return previousValue + currentValue.length;
  },
  0
);
```
Do you think the array "valuesForLengthOfAllAnimalNames" is necessary for better understanding? If so nvm :)

Reply:
I completely see your point that we should get rid of the `valuesForLengthOfAllAnimalNames` variable; I have implemented it to make the very first usage of `reduce` easier for the students.

I still think that your solution is quite heavy to understand (although it would be a proper one!).

This is why I wrote hints for two (rather easy) options, using the `.length` property either with or without `map` in the first place.